iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/execution/config/executor/ClusterConfigTaskExecutor.java

```diff
@@ -1663,10 +1663,12 @@ public SettableFuture<ConfigTaskResult> alterPipe(AlterPipeStatement alterPipeSt
 
     // Validate before alteration
     try {
-      if (!alterPipeStatement.getProcessorAttributes().isEmpty()) {
+      if (!alterPipeStatement.getProcessorAttributes().isEmpty()
+          && alterPipeStatement.isReplaceAllProcessorAttributes()) {
         PipeAgent.plugin().validateProcessor(alterPipeStatement.getProcessorAttributes());
```

Comment:
I think it's better to try validate the "modify" as well if the pipeMeta persists in the PipeTaskMeta. If not, you can do nothing here.